src/helm/benchmark/scenarios/melt_scenarios.py

```diff
@@ -343,3 +351,947 @@ def __init__(self, **kwargs):
+        math_symbols = MATH_SYMBOLS
+
+        for sample_idx in range(self.num_train_instances + self.num_val_instances + self.num_test_instances):
+            # Sample rule symbols
+            sampled_rule_symbols = list(self.rng.choice(rule_symbols, size=self.rng.randint(2, 4)))
+            sampled_rule_symbols_set = sorted(list(set(sampled_rule_symbols)))  # sorted to make it deterministic
+
+            # Sample math symbols
+            sampled_math_symbols = list(self.rng.choice(math_symbols, size=self.rng.randint(2, 4)))
+
+            # generate the pattern
+            pattern = self.gen_pattern(sampled_math_symbols, sampled_rule_symbols)
+
+            # generate one substitution
+            substitute_dict, substitute_dict_str = self.gen_subst(sampled_rule_symbols_set, tokens)
+            result = pattern_subst(pattern, sampled_rule_symbols_set, substitute_dict)
+
+            # generate another substitution
+            substitute_dict_2, _ = self.gen_subst(sampled_rule_symbols_set, tokens)
+            result_2 = pattern_subst(pattern, sampled_rule_symbols_set, substitute_dict_2)
+
+            result_string = " ".join(result)
+            pattern_string = " ".join(pattern)
+
+            src: str
+            tgt: str
+            if self.mode == "induction":
+                result_string_2 = " ".join(result_2)
+                src = f"Hai kết quả: {result_string} | {result_string_2}"
+                tgt = f"Quy luật: {pattern_string}"
+            elif self.mode == "variable_substitution":
+                src = f"Các quy luật: {pattern_string} | Thay thế: {substitute_dict_str}"
+                tgt = " ".join(result)
+            elif self.mode == "pattern_match":
+                # we sample 3 other pattern strings as negatives for patterns matching.
+                other_patterns = [
+                    " ".join(self.gen_pattern(sampled_math_symbols, sampled_rule_symbols_set)) for _ in range(3)
+                ]
+                all_patterns = other_patterns + [pattern_string]
+                self.rng.shuffle(all_patterns)
+                all_pattern_string = " | ".join(all_patterns)
+                src = f"Các quy luật: {all_pattern_string} | Kết quả: {result_string}"
+                tgt = pattern_string
+            else:
+                raise ValueError(f"Invalid mode: {self.mode}")
+
+            split: str
+            if sample_idx < self.num_train_instances:
+                split = TRAIN_SPLIT
+            elif sample_idx < self.num_train_instances + self.num_val_instances:
+                split = VALID_SPLIT
+            else:
+                split = TEST_SPLIT
+
+            instance = Instance(
+                input=Input(text=src),
+                references=[Reference(Output(text=tgt), tags=[CORRECT_TAG])],
+                split=split,
+            )
+            instances.append(instance)
+
+        return instances
+
+
+def remove_boxed(string: str) -> Optional[str]:
```

Comment:
The methods functions between `removed_boxed` and `is_equiv_chain_of_thought` are unused; please remove them.